processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/CommunicationDetailsDeriver.java

```diff
@@ -159,4 +164,21 @@ public CommunicationDetails processOneToOne(String tenantId, Span item) throws R
         return ret;
     }
 
+    private void setServiceNames(CommunicationDetails communicationDetails, Span span, ProducerInfo producerInfo) {
+
+        // service name from server span
+        if (span.service() != null) {
```

Comment:
May be better to move this to the calling method, as in the FragmentCompletionTimeDriver, as the other code isn't required?